### DISCUSSION TITLE
Query doesn't refresh anymore on windows refocus

### DISCUSSION THREAD

**Discussion by vjau** [TanStack/query#8372] [Q&A] [answered]
Hi,
i have been using React Query successfully for some years on multiple projects, so i thought i had a pretty good understanding of it.
Now, i have started a new project with trpc (as my other projects), and i have a strange problem : my queries don't refetch anymore on windows refocus. 
I know there have been recent changes with v5 about refocus behaviour, but 1) my other apps are still working fine 2) when i listen to the isVisible event with the focus manager, it seems to fire at the right times, at least with tab changes (windows changes is another story).
With this very simple code 
```ts
export function Test() {
  const user = trpc.user.getCurrent.useQuery().data;
  console.log("user in Test", user);
  return <div>{user?.fullname}</div>;
}
```
The query doesn't ever refetch.
Devtools shows that is is stale, but it never refetch.
For it to refetch i have to unmount/remount the component, but windows refocus never does anything.

I just worked 3 hours on this without making any progress, what could cause this ?

Thank you

**Comment by TkDodo** [comment]
I can only say that it should work, and I can take a look if you can reproduce this in a stackblitz or codesandbox

**Reply by vjau** [reply]
Thank you, i am working on a repro.

**Comment by vjau** [comment, marked as answer]
Ok, i have found my mistake (after too many hours on this).
If you wrap your app like this with trpc + query 
```ts
<trpc.Provider client={trpcClient} queryClient={queryClient}>
  <App/>
</trpc.Provider>
```
(so forgetting to wrap a second time with `<QueryClientProvider client={queryClient}>`
It will seems to work. You won't receive any error message. queries will run, invalidate, rerun when invalidated or remounted. But not refetch on refocus. 
I will see that with them, thank you.
